Test suite fails for scitokens-1.6.2 on SL7: crytography does not support secp192r1

On SL 7.9 the test suite for scitokens-1.6.2 fails because the cryptography library is too old:

```
$ python3 -m pytest tests/ -v -x --lf
========================================= test session starts =========================================
platform linux -- Python 3.6.8, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /usr/bin/python3
cachedir: .cache
run-last-failure: rerun last 2 failures
rootdir: /home/duncan.macleod/git/scitokens, inifile:
plugins: xdist-0.0.0, mock-1.10.4, cov-2.5.1
collected 45 items

tests/test_create_scitoken.py::TestCreation::test_autodetect_keytype FAILEDsource:28: DeprecationWarning: invalid escape sequence \w


============================================== FAILURES ===============================================
________________________________ TestCreation.test_autodetect_keytype _________________________________

self = <test_create_scitoken.TestCreation testMethod=test_autodetect_keytype>

    def test_autodetect_keytype(self):
        """
            Test the autodetection of the key type
            """
        private_key = generate_private_key(
            public_exponent=65537,
            key_size=2048,
            backend=default_backend()
        )

        ec_private_key = ec.generate_private_key(
            ec.SECP256R1(), default_backend()
        )

        # Test when we give it the wrong algorithm type
        with self.assertRaises(scitokens.scitokens.UnsupportedKeyException):
            token = scitokens.SciToken(key = private_key, algorithm="ES256")

        # Test when we give it the wrong algorithm type
        with self.assertRaises(scitokens.scitokens.UnsupportedKeyException):
            token = scitokens.SciToken(key = ec_private_key, algorithm="RS256")

        # Test when we give an unsupported algorithm
        unsupported_private_key = ec.generate_private_key(
>           ec.SECP192R1(), default_backend()
        )

tests/test_create_scitoken.py:291:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib64/python3.6/site-packages/cryptography/hazmat/primitives/asymmetric/ec.py:287: in generate_private_key
    return backend.generate_elliptic_curve_private_key(curve)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cryptography.hazmat.backends.openssl.backend.Backend object at 0x7fec8cc16828>
curve = <cryptography.hazmat.primitives.asymmetric.ec.SECP192R1 object at 0x7fec8cc78da0>

    def generate_elliptic_curve_private_key(self, curve):
        """
            Generate a new private key on the named curve.
            """

        if self.elliptic_curve_supported(curve):
            curve_nid = self._elliptic_curve_to_nid(curve)

            ec_cdata = self._lib.EC_KEY_new_by_curve_name(curve_nid)
            self.openssl_assert(ec_cdata != self._ffi.NULL)
            ec_cdata = self._ffi.gc(ec_cdata, self._lib.EC_KEY_free)

            res = self._lib.EC_KEY_generate_key(ec_cdata)
            self.openssl_assert(res == 1)

            evp_pkey = self._ec_cdata_to_evp_pkey(ec_cdata)

            return _EllipticCurvePrivateKey(self, ec_cdata, evp_pkey)
        else:
            raise UnsupportedAlgorithm(
                "Backend object does not support {0}.".format(curve.name),
>               _Reasons.UNSUPPORTED_ELLIPTIC_CURVE
            )
E           cryptography.exceptions.UnsupportedAlgorithm: Backend object does not support secp192r1.

/usr/lib64/python3.6/site-packages/cryptography/hazmat/backends/openssl/backend.py:1346: UnsupportedAlgorithm
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================== 1 failed, 43 deselected in 0.75 seconds ===============================
```

Ungrading to cryptography >=2.4.1 solves this issue, I think. Should the relevant test be decorated with `unittest.skipIf(cryptography_version < "2.4.1")`?

--- a/tests/test_create_scitoken.py
+++ b/tests/test_create_scitoken.py
@@ -288,7 +288,7 @@ class TestCreation(unittest.TestCase):
 
         # Test when we give an unsupported algorithm
         unsupported_private_key = ec.generate_private_key(
-            ec.SECP192R1(), default_backend()
+            ec.SECP521R1(), default_backend()
         )
         with self.assertRaises(scitokens.scitokens.UnsupportedKeyException):
             token = scitokens.SciToken(key = unsupported_private_key)

